src/plugin/svipc/sysvipc.cpp

```diff
@@ -430,6 +430,37 @@ SysVIPC::serialize(jalib::JBinarySerializer &o)
   _virtIdTable.serialize(o);
 }
 
+int
+SysVShm::virtualToRealKey(key_t k)
+{
+  if (_keyMap.find(k) != _keyMap.end()) {
+    return _keyMap[k];
+  } else {
+    int realId = SharedData::getRealIPCId(SYSV_SHM_KEY, k);
+    updateKeyMapping(k, realId);
```

Comment:
Shall we update the mapping only if `realID != -1`? Just to avoid inserting a useless entry.